utils/gyb_syntax_support/__init__.py

```diff
@@ -141,5 +143,28 @@ def dedented_lines(description):
     return textwrap.dedent(description).split('\n')
 
 
+def hash_syntax_node(node):
+    # Hash into the syntax name and serialization code
+    result = hash((node.name, str(get_serialization_code(node.syntax_kind))))
+    for child in node.children:
+        # Hash into the expected child syntax
+        result = hash((result, child.syntax_kind))
+        # Hash into the child name
+        result = hash((result, child.name))
+        # Hash into whether the child is optional
+        result = hash((result, str(child.is_optional)))
+    return result
+
+
+def hash_token_syntax(token):
+    # Hash into the token name and serialization code
+    return hash((token.name, str(token.serialization_code)))
+
+
 def calculate_node_hash():
-    return "abc"
+    result = 0
+    for node in SYNTAX_NODES:
+        result = hash((result, hash_syntax_node(node=node)))
```

Comment:
Recommend `hash_syntax_node(node=node)` -> `hash_syntax_node(node)`, named arguments are not required in python, you usually put them for readability when there's a lot of them, but it's not offering any value here
Also remove named argument for `hash_token_syntax(token=token)` below.